app/src/main/kotlin/io/homeassistant/companion/android/controls/HaControlsProviderService.kt

```diff
@@ -265,7 +281,8 @@ class HaControlsProviderService : ControlsProviderService() {
                         systemId = it,
                         entityId = entityId,
                         serverId = serverId,
-                        area = getAreaForEntity(entity.entityId, serverId)
+                        area = getAreaForEntity(entity.entityId, serverId),
+                        splitMultiServerIntoStructure = splitMultiServersIntoStructures
```

Comment:
A `serverName` is not supplied here so it will never split, you can remove this line to simplify the code a bit.